lib/api/objectHead.js

```diff
@@ -124,10 +124,12 @@ function objectHead(authInfo, request, log, callback) {
                     return callback(errors.BadRequest, corsHeaders);
                 }
                 const partSize = getPartSize(objMD, partNumber);
-                if (!partSize) {
+                const isEmptyObject = objLength === 0;
+                if (!partSize && !isEmptyObject) {
```

Comment:
Maybe we should additionally enforce that the requested part is 1 in the empty object case, and return 416 otherwise (to match AWS behavior which is akin to considering an empty object as having a single empty part)